components/config/definition.rst

```diff
@@ -200,43 +200,158 @@ Array Node Options
 
 Before defining the children of an array node, you can provide options like:
 
-``useAttributeAsKey()``
-    Provide the name of a child node, whose value should be used as the key in the resulting array.
-``requiresAtLeastOneElement()``
-    There should be at least one element in the array (works only when ``isRequired()`` is also
-    called).
 ``addDefaultsIfNotSet()``
-    If any child nodes have default values, use them if explicit values haven't been provided.
+    If any child nodes have default values, use them if explicit values haven't
+    been provided.
+``requiresAtLeastOneElement()``
+    There should be at least one element in the array (works only when
+    ``isRequired()`` is also called).
+``useAttributeAsKey()``
+    Provide the name of a child node, whose value should be used as the key in
+    the resulting array. This method also defines the way config array keys are
+    treated, as explained in the following example.
 
-An example of this::
+A basic prototyped array configuration can be defined as follows::
 
-    $rootNode
+    $node
+        ->fixXmlConfig('driver')
         ->children()
-            ->arrayNode('parameters')
-                ->isRequired()
-                ->requiresAtLeastOneElement()
-                ->useAttributeAsKey('name')
+            ->arrayNode('drivers')
+                ->prototype('scalar')->end()
+            ->end()
+        ->end()
+    ;
+
+When using the following YAML configuration:
+
+.. code-block:: yaml
+
+    drivers: ['mysql', 'sqlite']
+
```

Comment:
yes it is, because this is exactly the same in YAML, so the Config component receives the same input.